### PR TITLE
FST benchmark

### PR DESCRIPTION
**Pull request by karthikeyann**
Depends on #11242 Feature/finite state transducer 

Benchmark for Finite State Transducer
parse and identify JSON symbols
- [x] FST with output, output index, output str
- [x] FST without output index
- [x] FST without, output
- [x] FST without output str

Look into https://github.com/elstehle/cudf/pull/1 for files modified only in this PR (i.e excluding parent depending PR)

### CODE REVIEW

**Review by PointKernel — APPROVED** [review]
LGTM

(Focused on `cpp/benchmarks/io/fst.cu` onlly)

**Review by elstehle — APPROVED** [review]
Just one minor comment. Otherwise LGTM.
Also, thanks for adding the `DiscardOutputIterator` versions!

File: cpp/benchmarks/io/fst.cu [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+#include <cstdlib>
+
+namespace cudf {
+namespace {
+auto make_test_json_data(nvbench::state& state)
+{
+  auto const string_size{size_type(state.get_int64("string_size"))};
+  // auto const depth{size_type(state.get_int64("depth"))};
+  // Basic JSON structure.
+  // number, string, boolean, Array, { "str_key": JSON obj }, null.
```
[the remark this answers is not in the document]
Comment (karthikeyann) [reply cpp/benchmarks/io/fst.cu:46]: Yes. supposed 
```suggestion
```

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**